clients/fides-js/src/components/tcf/TcfPurposes.tsx

```diff
@@ -155,6 +160,57 @@ const TcfPurposes = ({
     enabledCustomPurposeConsentIds,
   ]);
 
+  /**
+   * Handles toggling of purposes and special purposes in the TCF interface.
+   *
+   * Takes a modelType (e.g., purposesConsent, specialPurposes) that determines how
+   * the change will be persisted, a list of newly enabled IDs, the purpose/notice
+   * being toggled (which can be a standard TCF purpose, custom privacy notice, or
+   * special purpose), and details about what triggered the toggle.
+   */
+  const handleToggle = (
+    modelType: keyof EnabledIds,
+    newEnabledIds: string[],
+    item:
+      | PurposeRecord
+      | PrivacyNoticeWithBestTranslation
+      | TCFSpecialPurposeRecord,
+    triggerDetails: FidesEventDetailsTrigger,
+  ) => {
+    // Determine the preference being changed based on the model type:
+    // - customPurposesConsent -> notice
+    // - purposesConsent -> tcf_purpose_consent
+    // - purposesLegint/specialPurposes -> tcf_purpose_legitimate_interest
+    //
+    // NOTE: TCF purposes don't have equivalent notice "keys" today, but we
+    // expect to add those in the future. When we do, we'll prefix like this:
+    // "tcf_purpose_consent_1", "tcf_purpose_legitimate_interest_2", etc.
```

Comment:
Note this! This is the first place we've exposed a TCF purpose id in our FidesJS SDK and I didn't want to ship vanilla numbers like `1` or `2` because they lack context. Therefore I've prefixed them here with the preference type so you can separate between purpose consents, special features, etc.